src/chainrules.jl

```diff
@@ -34,27 +34,36 @@ function ChainRulesCore.rrule(::typeof(logsumexp), x::AbstractArray{<:Real}; dim
     return Ω, logsumexp_pullback
 end
 
-function ChainRulesCore.frule(
-    (_, _, Δx), ::typeof(softmax!), r::AbstractArray{<:Real}, x::AbstractArray{<:Real},
-)
-    softmax!(r, x)
-    _Δx = reshape(Δx, size(r))
-    Δr = r .* (_Δx .- LinearAlgebra.dot(r, _Δx))
-    return r, Δr
+# no rules for mutating functions currently:
+# https://juliadiff.org/ChainRulesCore.jl/stable/writing_good_rules.html#Which-functions-need-rules?
+function ChainRulesCore.frule((_, Δx), ::typeof(softmax), x::AbstractArray{<:Real}; dims=:)
+    Ω = softmax(x; dims=dims)
+    ΔΩ = if dims === Colon()
```

Comment:
```suggestion
    ΔΩ = if dims === :
```

Reply:
This is not allowed (https://github.com/JuliaStats/LogExpFunctions.jl/pull/28/checks?check_run_id=3576281866#step:6:97), therefore I'll revert the suggestion.

Reply:
And the suggestion below.

Reply:
Ah, I can never remember when this works or not...